skyportal/handlers/api/source.py

```diff
@@ -1105,25 +1158,32 @@ class Validator(Schema):
                         ).filter(Annotation.obj_id == obj.id),
                         key=lambda x: x.origin,
                     )
-                obj_list[-1]["last_detected_at"] = obj.last_detected_at(
-                    self.current_user
-                )
-                obj_list[-1]["last_detected_mag"] = obj.last_detected_mag(
-                    self.current_user
-                )
-                obj_list[-1]["peak_detected_at"] = obj.peak_detected_at(
-                    self.current_user
+
+                obj_list[-1]["last_detected_at"] = (
+                    (last_detected_at - last_detected_at.utcoffset()).replace(
+                        tzinfo=UTC
+                    )
+                    if last_detected_at
+                    else None
```

Comment:
The SQL expression version of the `last_detected_at` and `peak_detected_at` returns the values in local time for some reason, so we just change it back to UTC again here.